modules/beta-private-cluster/main.tf

```diff
@@ -57,231 +60,63 @@ locals {
-      0,
-    )
-  }
-
-  cluster_type_output_location = {
-    regional = element(concat(google_container_cluster.primary.*.region, [""]), 0)
-    zonal = element(
-      concat(google_container_cluster.zonal_primary.*.zone, [""]),
-      0,
-    )
-  }
-
-  cluster_type_output_region = {
-    regional = element(concat(google_container_cluster.primary.*.region, [""]), 0)
-    zonal    = var.region
-  }
-
-  cluster_type_output_regional_zones = flatten(google_container_cluster.primary.*.node_locations)
-  cluster_type_output_zonal_zones    = local.zone_count > 1 ? slice(var.zones, 1, local.zone_count) : []
-
-  cluster_type_output_zones = {
-    regional = local.cluster_type_output_regional_zones
-    zonal = concat(
-      google_container_cluster.zonal_primary.*.zone,
-      local.cluster_type_output_zonal_zones,
-    )
-  }
-
-  cluster_type_output_endpoint = {
-    regional = var.deploy_using_private_endpoint ? element(concat(google_container_cluster.primary.*.private_cluster_config.0.private_endpoint, [""]), 0) : element(concat(google_container_cluster.primary.*.endpoint, [""]), 0)
-
-    zonal = var.deploy_using_private_endpoint ? element(concat(google_container_cluster.zonal_primary.*.private_cluster_config.0.private_endpoint, [""]), 0) : element(concat(google_container_cluster.zonal_primary.*.endpoint, [""]), 0)
-  }
-
-  cluster_type_output_master_auth = {
-    regional = concat(google_container_cluster.primary.*.master_auth, [])
-    zonal    = concat(google_container_cluster.zonal_primary.*.master_auth, [])
-  }
 
-  cluster_type_output_master_version = {
-    regional = element(
-      concat(google_container_cluster.primary.*.master_version, [""]),
-      0,
-    )
-    zonal = element(
-      concat(
-        google_container_cluster.zonal_primary.*.master_version,
-        [""],
-      ),
-      0,
-    )
-  }
+  cluster_output_name           = concat(google_container_cluster.primary.*.name, [""])[0]
+  cluster_output_location       = concat(google_container_cluster.primary.*.location, [""])[0]
+  cluster_output_region         = concat(google_container_cluster.primary.*.region, [""])[0]
+  cluster_output_regional_zones = flatten(google_container_cluster.primary.*.node_locations)
+  cluster_output_zonal_zones    = local.zone_count > 1 ? slice(var.zones, 1, local.zone_count) : []
+  cluster_output_zones          = local.cluster_output_regional_zones
 
-  cluster_type_output_min_master_version = {
-    regional = element(
-      concat(google_container_cluster.primary.*.min_master_version, [""]),
-      0,
-    )
-    zonal = element(
-      concat(
-        google_container_cluster.zonal_primary.*.min_master_version,
-        [""],
-      ),
-      0,
-    )
-  }
+  cluster_output_endpoint = var.deploy_using_private_endpoint ? element(concat(google_container_cluster.primary.*.private_cluster_config.0.private_endpoint, [""]), 0) : element(concat(google_container_cluster.primary.*.endpoint, [""]), 0)
```

Comment:
Here we can change notation too 
`cluster_output_endpoint = var.deploy_using_private_endpoint ? concat(google_container_cluster.primary.*.private_cluster_config.0.private_endpoint, [""])[0] : concat(google_container_cluster.primary.*.endpoint, [""])[0]`

Reply:
will search for all element() in a code as well